### ISSUE TITLE
[bat error]: invalid digit found in string

### ISSUE THREAD

**Issue by orhun** [kyu08/fzf-make#53]
Hey,

I just tried out `fzf-make` on [Shuttle's Makefile](https://github.com/shuttle-hq/shuttle/blob/main/Makefile) and got this error.

![cap](https://github.com/kyu08/fzf-make/assets/24392180/72f91ba9-5662-4da3-bf19-f20fe084bd68)

Also, I'm getting a different error for the last entry:

![cap](https://github.com/kyu08/fzf-make/assets/24392180/5b1c439c-78eb-4fa0-a5e5-ffe4694cf964)

Thanks for this tool!

**Comment by kyu08** [comment]
Thank you for using fzf-make and reporting bugs!
I'll investigate soon.

**Comment by kyu08** [comment]
Somehow, I couldn't reproduce it in my environment. (macOS: 13.4, bat version: 0.23.0, fzf-make version: v0.6.0)

<img width="1505" alt=" 2023-08-19 at 2 55 56" src="https://github.com/kyu08/fzf-make/assets/49891479/4b503e48-086a-4708-97bc-1aff6b8031bc">
<img width="1505" alt=" 2023-08-19 at 2 56 02" src="https://github.com/kyu08/fzf-make/assets/49891479/a4a472eb-7618-4564-b094-1aa6a17cb983">

So I need your environmental information.

Could you tell me following things?

- OS and its version
- bat version
- fzf-make version (If you find out. Maybe not important)

If you use linux, perhaps my build method is bad.
I just run `cargo build --verbose --release` on my macOS machine.

**Comment by orhun** [comment]
OS: Arch Linux
Bat: bat 0.23.0 (871abd27)
fzf-make: `0.6.0` (installed from the [AUR](https://aur.archlinux.org/packages/fzf-make))

I just tried out the latest git version from this repo and got the same error :/

**Comment by kyu08** [comment]
@orhun 
Thank you for letting me know your environment!
Hmm...

For now I have no idea, but I'll do some research using docker image of Arch Linux.
Please wait for me to contact you when I make progress.
If you find out anything, I'd appreciate you letting me know in this issue.

**Comment by orhun** [comment]
I think it fails because `skim` is unable to run this command:

https://github.com/kyu08/fzf-make/blob/f132a72228b5023baa581ca1e352bbe5f0bb92fa/src/fuzzy_finder/fuzzy_finder.rs#L33-L47

**Comment by orhun** [comment]
I found the issue, `grep` behaves differently on macOS and Linux I guess.

If you don't provide multiple files, `grep` doesn't print out the file name on Linux:

```sh
$ grep -rnE '^clean\s*:' Makefile

117:clean:
```

Workaround: https://stackoverflow.com/questions/15432156/display-filename-before-matching-line

```sh
$ grep -rnE '^clean\s*:' Makefile /dev/null

Makefile:117:clean:
```

There is also `--with-filename` option but I'm not sure if it is supported on MacOS.

I created #56 as a fix. Everything works as expected now:

![cap](https://github.com/kyu08/fzf-make/assets/24392180/b7b82ee9-ce44-4f8d-8fe8-66581f736403)
